pkg/config/template.go

```diff
@@ -47,3 +62,68 @@ func getTemplates(templatePath string) (CiTemplates, error) {
 func isYAML(file string, info os.FileInfo) bool {
 	return !info.IsDir() && (filepath.Ext(file) == ".yaml" || filepath.Ext(file) == ".yml")
 }
+
+// TemplateCMController holds the details needed for the configmap controller
+type TemplateCMController struct {
+	cmclient  corev1.ConfigMapInterface
+	prNumber  int
+	logger    *logrus.Entry
+	templates map[string]*templateapi.Template
+}
+
+// NewTemplateCMController creates a new TemplateCMController
+func NewTemplateCMController(cmclient corev1.ConfigMapInterface, prNumber int, logger *logrus.Entry, templates map[string]*templateapi.Template) *TemplateCMController {
+	return &TemplateCMController{
+		cmclient:  cmclient,
+		prNumber:  prNumber,
+		logger:    logger,
+		templates: templates,
+	}
+}
+
+// CreateCMTemplates creates configMaps for all the changed templates.
+func (c *TemplateCMController) CreateCMTemplates() error {
+	var errors []error
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	for filename, template := range c.templates {
+		buf := new(bytes.Buffer)
+
+		err := s.Encode(template, buf)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		cmName := GetTempCMName(c.prNumber, template.Name)
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cmName,
+				Labels: map[string]string{
+					"ci.openshift.org/rehearse": "rehearse-temporary-cm",
+				},
+			},
+			Data: map[string]string{filename: buf.String()},
+		}
+
+		c.logger.WithField("template-name", template.Name).Info("creating rehearsal configMap for template")
+		if _, err := c.cmclient.Create(cm); err != nil && !kerrors.IsAlreadyExists(err) {
```

Comment:
What is the reasoning behind the suppressing the `AlreadyExists` error? That does not seem right. If the CM exists, it may contain some stale old content from some previous run. That would be a problem, we need the rehearsals to use the content that came with the PR.

Reply:
Shall we remove it if alreadyExists first?

Reply:
Maybe, if already exists, just update it?

Reply:
Updating the content sounds good, should not break anything.